Dockerfile

```diff
@@ -0,0 +1,28 @@
+# syntax=docker/dockerfile:1
+FROM ruby:2.7.4
+
+# Set up working directory
+RUN mkdir /code
+WORKDIR /code
+COPY . /code
+
+# Set up for production mode build
+ENV RAILS_ENV production
+ENV RAILS_SERVE_STATIC_FILES true
+ENV RAILS_LOG_TO_STDOUT true
+
+# Install gem dependencies
+RUN bundle install
+# Install Yarn.
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+RUN apt-get update && apt-get install -y yarn
+# Run yarn install to install JavaScript dependencies.
+RUN yarn install --check-files
+
+# Compile for production build
+RUN yarn build
```

Comment:
```suggestion
WORKDIR /usr/src/app

# Set environment for production mode build
ENV RAILS_ENV production
ENV RAILS_SERVE_STATIC_FILES true
ENV RAILS_LOG_TO_STDOUT true

# Operating system dependencies
RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
RUN apt-get update && apt-get install -y yarn

# Install Ruby Dependencies
COPY Gemfile Gemfile.lock ./
RUN bundle install

# Install JavaScript Dependencies
COPY package.json yarn.lock ./
RUN yarn install --check-files
# Compile for production build
RUN yarn build

# Set up working directory
COPY . .
```

We can get a lot of performance improvement in the docker build by leveraging the Docker build cache. By putting the steps _least likely to change_ first we can reuse existing images for many of the time consuming first steps.

Helpful References:

* [Docker Docs: Dockerfile Best Practices#Leverage Build Cache](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache)
* [Dockerfile Ruby Best Practices: Place the least likely to change commands at the top](https://lipanski.com/posts/dockerfile-ruby-best-practices#6-place-the-least-likely-to-change-commands-at-the-top)
* [Example Inferno Dockerfile which pulls in Gemfile first](https://github.com/onc-healthit/inferno/blob/main/Dockerfile#L5-L25)

Reply:
I had to move the `RUN yarn build` step down below the `COPY . .` step, otherwise Docker was unable to access the Rakefile. Other than that, this update definitely increases build speed!